### PR TITLE
Improve the multiscalar multiplication precomputation step.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Avoid field operations since they can be slow.

While it does not appear to have a significant effect for the BLS curve, but still has non-trivial effect, it might matter more with other curves.

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: rust-src/concordium_base/src/curve_arithmetic/mod.rs [hunk 1]
```diff
@@ -212,46 +212,70 @@ pub fn multiexp_worker<C: Curve, X: Borrow<C>>(
 pub fn multiexp_worker_given_table<C: Curve>(
     exps: &[C::Scalar],
     table: &[Vec<C>],
-    window_size: usize,
+    window: usize,
 ) -> C {
     // Compute the wnaf
-
+    let window_size = window + 1;
```
[the remark this answers is not in the document]
Comment (hamiidreza) [reply rust-src/concordium_base/src/curve_arithmetic/mod.rs:218]: The code here makes sense to me. What we implement here is the NAF method with width "window+1" which has a different width parameter compared to dalek NAF with width "window". Calling it "window_size" looks a bit bit confusing and inconsistent with the rest of the code, though! I would call it something that reflects that this is 1 plus the window parameter, e.g., "window_plus1".

File: rust-src/concordium_base/src/curve_arithmetic/mod.rs [hunk 2]
```diff
@@ -212,46 +212,70 @@ pub fn multiexp_worker<C: Curve, X: Borrow<C>>(
...
     let k = exps.len();
     // assert_eq!(gs.len(), k);
-    assert!(window_size >= 1);
```
Comment (hamiidreza) [inline rust-src/concordium_base/src/curve_arithmetic/mod.rs:221]: Why is this commented? Should it not be `assert!(window_size >= 2);` now?

File: rust-src/concordium_base/src/curve_arithmetic/mod.rs [hunk 3]
```diff
@@ -212,46 +212,70 @@ pub fn multiexp_worker<C: Curve, X: Borrow<C>>(
...
     assert!(window_size < 62);
```
Comment (hamiidreza) [inline rust-src/concordium_base/src/curve_arithmetic/mod.rs:221]: I think for consistency with the rest of the code, this should also be `assert!(window_size < 63);`.

Comment (abizjak) [reply rust-src/concordium_base/src/curve_arithmetic/mod.rs:221]: Ok.

File: rust-src/concordium_base/src/curve_arithmetic/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (abizjak) [reply rust-src/concordium_base/src/curve_arithmetic/mod.rs:218]: Ok.

File: rust-src/concordium_base/src/curve_arithmetic/mod.rs (hunk 2 above)
Comment (abizjak) [reply rust-src/concordium_base/src/curve_arithmetic/mod.rs:221]: I reinstated it.

**Pull request merged**